panic in the  TestRenameTableIntermediateState

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4ac79d1]


goroutine 383949 [running]:

github.com/pingcap/tidb/ddl_test.TestRenameTableIntermediateState.func1(0xc09119a160?)

	ddl/table_test.go:559 +0xb1

github.com/pingcap/tidb/ddl/internal/callback.(*TestDDLCallback).OnJobUpdated(0xc090cf7080, 0xc09119a160?)

	ddl/internal/callback/callback.go:122 +0x1ea

github.com/pingcap/tidb/ddl.(*ddl).delivery2worker.func1()

	ddl/job_table.go:451 +0xdef

github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()

	util/wait_group_wrapper.go:154 +0x6b

created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run

	util/wait_group_wrapper.go:152 +0x8d

--

```

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_unit_test/detail/ghpr_unit_test/24468/pipeline

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/table_test.go
+++ b/ddl/table_test.go
@@ -553,9 +553,11 @@ func TestRenameTableIntermediateState(t *testing.T) {
 		hook := &callback.TestDDLCallback{Do: dom}
 		runInsert := false
 		fn := func(job *model.Job) {
-			if job.SchemaState == model.StatePublic && !runInsert && !t.Failed() {
+			if job.Type == model.ActionRenameTable &&
+				job.SchemaState == model.StatePublic && !runInsert && !t.Failed() {
 				_, err := tk2.Exec(tc.insertSQL)
 				if len(tc.errMsg) > 0 {
+					assert.NotNil(t, err)
 					assert.Equal(t, tc.errMsg, err.Error())
 				} else {
 					assert.NoError(t, err)

